### ISSUE TITLE
[BGFIX] Many wallpapers wont load: "error C1503: undefined variable "M_PI""

### ISSUE THREAD

**Issue by Arna13** [Almamu/linux-wallpaperengine#103]
**Wallpaper Engine Background(s)**

Background IDs Ive tried:
    - 2463435445
    - 2663183306
    - 909458039
    - 1833566210

**Console output (2463435445)**

```Detected scene.pkg file at /home/arna13/.local/share/Steam/steamapps/workshop/content/431960/2463435445/scene.pkg. Adding to list of searchable paths
Found wallpaper engine's assets at /home/arna13/.local/share/Steam/steamapps/common/wallpaper_engine/assets
terminate called after throwing an instance of 'std::runtime_error'
  what():  0(105) : error C1503: undefined variable "M_PI"
0(146) : error C1503: undefined variable "M_PI"

Compiled source code:
#version 120
#define highp
#define mediump
#define lowp
#define mul(x, y) (y * x)
#define max(x, y) max (y, x)
#define frac fract
#define CAST2(x) (vec2(x))
#define CAST3(x) (vec3(x))
#define CAST4(x) (vec4(x))
#define CAST3X3(x) (mat3(x))
#define saturate(x) (clamp(x, 0.0, 1.0))
#define texSample2D texture2D
#define texSample2DLod texture2D
#define atan2 atan
#define ddx dFdx
#define ddy(x) dFdy(-(x))
#define GLSL 1
#define HLSL 1
#define LIGHTING 0
#define NORMALMAP 1
#define PBRMASKS 1
#define REFLECTION 0
#define VERSION 2




uniform sampler2D g_Texture0; // {"label":"ui_editor_properties_albedo","nonremovable":true}

#ifndef VERSION
uniform float g_Brightness; // {"material":"Brightness","label":"ui_editor_properties_brightness","default":1,"range":[0,2]}
uniform float g_UserAlpha; // {"material":"Alpha","label":"ui_editor_properties_alpha","default":1,"range":[0,1]}
#else
uniform float g_Alpha;
uniform vec3 g_Color;
#endif

#if PBRMASKS
varying vec4 v_TexCoord;
#else
varying vec2 v_TexCoord;
#endif

#if LIGHTING || REFLECTION
uniform sampler2D g_Texture1; // {"label":"ui_editor_properties_normal_map","combo":"NORMALMAP","format":"rg88","mode":"normal","requireany":true,"require":{"LIGHTING":1,"REFLECTION":1}}
uniform sampler2D g_Texture2; // {"combo":"PBRMASKS","mode":"opacitymask","paintdefaultcolor":"0 0 0 1","components":[{"label":"ui_editor_properties_metallic_map","combo":"METALLIC_MAP"},{"label":"ui_editor_properties_roughness_map","combo":"ROUGHNESS_MAP"},{"label":"ui_editor_properties_reflection_map","combo":"REFLECTION_MAP"}],"requireany":true,"require":{"LIGHTING":1,"REFLECTION":1}}
uniform float g_Roughness; // {"material":"roughness","label":"ui_editor_properties_roughness","default":0.5,"range":[0,1],"nobindings":true}
uniform float g_Metallic; // {"material":"metallic","label":"ui_editor_properties_metallic","default":0.5,"range":[0,1],"nobindings":true}
#endif

#if LIGHTING
uniform vec4 g_LightsColorPremultiplied[3];
uniform vec3 g_LightAmbientColor;

varying vec4 v_Light0DirectionL3X;
varying vec4 v_Light1DirectionL3Y;
varying vec4 v_Light2DirectionL3Z;
#endif

// Local space normal direction and precomputed tangent space in light vectors
#if (LIGHTING || REFLECTION) && NORMALMAP == 0
// World space normal direction without normal map
varying vec3 v_Normal;
#endif

#if REFLECTION && NORMALMAP
uniform vec3 g_Screen;
uniform sampler2D g_Texture3; // {"hidden":true,"default":"_rt_MipMappedFrameBuffer"}
uniform float g_Reflectivity; // {"material":"reflectivity","label":"ui_editor_properties_reflectivity","default":1,"range":[0,1],"nobindings":true}
uniform float g_Texture3MipMapInfo;

varying vec3 v_Tangent;
varying vec3 v_Bitangent;
varying vec3 v_ScreenPos;
#endif

#if BLENDMODE
uniform sampler2D g_Texture4; // {"hidden":true,"default":"_rt_FullFrameBuffer"}
varying vec3 v_ScreenCoord;
#endif



// begin of included from file common_pbr.h



vec3 FresnelSchlick(float lightTheta, vec3 baseReflectance)
{
	return baseReflectance + (1.0 - baseReflectance) * pow(max(1.0 - lightTheta, 0.001), 5.0);
}

float Distribution_GGX(vec3 N, vec3 H, float roughness)
{
	float rSqr = roughness * roughness;
	float rSqr2 = rSqr * rSqr;

	float NH = max(dot(N, H), 0.0);
	float NH2 = NH * NH;

	float numerator = rSqr2;
	float denominator = (NH2 * (rSqr2 - 1.0) + 1.0);

	return numerator / (M_PI * denominator * denominator);
}

float Schlick_GGX(float NV, float roughness)
{
	float roughnessBase = roughness + 1.0;
	float roughnessScaled = (roughnessBase * roughnessBase) / 8.0;
	float denominator = NV * (1.0 - roughnessScaled) + roughnessScaled;
	return NV / denominator;
}

float GeoSmith(vec3 N, vec3 V, vec3 L, float roughness)
{
	float NV = max(dot(N, V), 0.001);
	float NL = max(dot(N, L), 0.001);
	return Schlick_GGX(NV, roughness) * Schlick_GGX(NL, roughness);
}

vec3 ComputePBRLight(vec3 normalVector, vec3 worldToLightVector, vec3 worldToViewVector,
	vec3 albedo, vec3 lightColor, vec3 baseReflectance, float roughness, float metallic)
{
	float distance = length(worldToLightVector);
	vec3 L = worldToLightVector / distance;
	vec3 H = normalize(worldToViewVector + L);
	vec3 N = normalVector;
	vec3 V = worldToViewVector;

	vec3 radiance = lightColor / (distance * distance);

	float NDF = Distribution_GGX(N, H, roughness);
	float G = GeoSmith(N, V, L, roughness);
	vec3 F = FresnelSchlick(max(dot(H, V), 0.0), baseReflectance);

	vec3 diffuse = CAST3(1.0) - F;
	diffuse *= 1.0 - metallic;

	vec3 numerator = NDF * G * F;
	float denominator = 4.0 * max(dot(N, V), 0.0) * max(dot(N, L), 0.0);
	vec3 specular = numerator / max(denominator, 0.001);

	float NL = max(dot(N, L), 0.0);
	return (diffuse * albedo / M_PI + specular) * radiance * NL;
}

// end of included from file common_pbr.h
// begin of included from file common_blending.h

vec4 Desaturate(vec3 color, float Desaturation)
{
	vec3 grayXfer = vec3(0.3, 0.59, 0.11);
	vec3 gray = CAST3(dot(grayXfer, color));
	return vec4(mix(color, gray, Desaturation), 1.0);
}

vec3 RGBToHSL(vec3 color)
{
	vec3 hsl;
	float fmin = min(min(color.r, color.g), color.b);
	float fmax = max(max(color.r, color.g), color.b);
	float delta = fmax - fmin;
	hsl.z = (fmax + fmin) / 2.0;

	if (delta == 0.0)
	{
		hsl.x = 0.0;
		hsl.y = 0.0;
	}
	else
	{
		if (hsl.z < 0.5)
			hsl.y = delta / (fmax + fmin);
		else
			hsl.y = delta / (2.0 - fmax - fmin);
		float deltaR = (((fmax - color.r) / 6.0) + (delta / 2.0)) / delta;
		float deltaG = (((fmax - color.g) / 6.0) + (delta / 2.0)) / delta;
		float deltaB = (((fmax - color.b) / 6.0) + (delta / 2.0)) / delta;
		if (color.r == fmax )
			hsl.x = deltaB - deltaG;
		else if (color.g == fmax)
			hsl.x = (1.0 / 3.0) + deltaR - deltaB;
		else if (color.b == fmax)
			hsl.x = (2.0 / 3.0) + deltaG - deltaR;

		if (hsl.x < 0.0)
			hsl.x += 1.0;
		else if (hsl.x > 1.0)
			hsl.x -= 1.0;
	}

	return hsl;
}

float HueToRGB(float f1, float f2, float hue)
{
	if (hue < 0.0)
		hue += 1.0;
	else if (hue > 1.0)
		hue -= 1.0;
	float res;
	if ((6.0 * hue) < 1.0)
		res = f1 + (f2 - f1) * 6.0 * hue;
	else if ((2.0 * hue) < 1.0)
		res = f2;
	else if ((3.0 * hue) < 2.0)
		res = f1 + (f2 - f1) * ((2.0 / 3.0) - hue) * 6.0;
	else
		res = f1;
	return res;
}

vec3 HSLToRGB(vec3 hsl)
{
	vec3 rgb;
	if (hsl.y == 0.0)
		rgb = CAST3(hsl.z);
	else
	{
		float f2;
		if (hsl.z < 0.5)
			f2 = hsl.z * (1.0 + hsl.y);
		else
			f2 = (hsl.z + hsl.y) - (hsl.y * hsl.z);
		float f1 = 2.0 * hsl.z - f2;
		rgb.r = HueToRGB(f1, f2, hsl.x + (1.0/3.0));
		rgb.g = HueToRGB(f1, f2, hsl.x);
		rgb.b= HueToRGB(f1, f2, hsl.x - (1.0/3.0));
	}
	
	return rgb;
}

vec3 ContrastSaturationBrightness(vec3 color, float brt, float sat, float con)
{
	const float AvgLumR = 0.5;
	const float AvgLumG = 0.5;
	const float AvgLumB = 0.5;
	
	const vec3 LumCoeff = vec3(0.2125, 0.7154, 0.0721);
	
	vec3 AvgLumin = vec3(AvgLumR, AvgLumG, AvgLumB);
	vec3 brtColor = color * brt;
	vec3 intensity = CAST3(dot(brtColor, LumCoeff));
	vec3 satColor = mix(intensity, brtColor, sat);
	vec3 conColor = mix(AvgLumin, satColor, con);
	return conColor;
}

#define BlendLinearDodgef(base, blend) (base + blend)
#define BlendLinearBurnf(base, blend) max(base + blend - 1.0, 0.0)
#define BlendLightenf(base, blend) max(blend, base)
#define BlendDarkenf(base, blend) min(blend, base)
#define BlendLinearLightf(base, blend) (blend < 0.5 ? BlendLinearBurnf(base, (2.0 * blend)) : BlendLinearDodgef(base, (2.0 * (blend - 0.5))))
#define BlendScreenf(base, blend) (1.0 - ((1.0 - base) * (1.0 - blend)))
#define BlendOverlayf(base, blend) (base < 0.5 ? (2.0 * base * blend) : (1.0 - 2.0 * (1.0 - base) * (1.0 - blend)))
#define BlendSoftLightf(base, blend) ((blend < 0.5) ? (2.0 * base * blend + base * base * (1.0 - 2.0 * blend)) : (sqrt(base) * (2.0 * blend - 1.0) + 2.0 * base * (1.0 - blend)))
#define BlendColorDodgef(base, blend) ((blend == 1.0) ? blend : min(base / (1.0 - blend), 1.0))
#define BlendColorBurnf(base, blend) ((blend == 0.0) ? blend : max((1.0 - ((1.0 - base) / blend)), 0.0))
#define BlendVividLightf(base, blend) ((blend < 0.5) ? BlendColorBurnf(base, (2.0 * blend)) : BlendColorDodgef(base, (2.0 * (blend - 0.5))))
#define BlendPinLightf(base, blend) ((blend < 0.5) ? BlendDarkenf(base, (2.0 * blend)) : BlendLightenf(base, (2.0 *(blend - 0.5))))
#define BlendHardMixf(base, blend) ((BlendVividLightf(base, blend) < 0.5) ? 0.0 : 1.0)
#define BlendReflectf(base, blend) ((blend == 1.0) ? blend : min(base * base / (1.0 - blend), 1.0))
#define BlendNormal(base, blend) (blend)
#define BlendLighten BlendLightenf
#define BlendDarken	BlendDarkenf
#define BlendMultiply(base, blend) (base * blend)
#define BlendAverage(base, blend) ((base + blend) / 2.0)
#define BlendAdd(base, blend) min(base + blend, CAST3(1.0))
#define BlendSubstract(base, blend) max(base + blend - CAST3(1.0), CAST3(0.0))
#define BlendDifference(base, blend) abs(base - blend)
#define BlendNegation(base, blend) (CAST3(1.0) - abs(CAST3(1.0) - base - blend))
#define BlendExclusion(base, blend) (base + blend - 2.0 * base * blend)
#define BlendScreen(base, blend) vec3(BlendScreenf(base.r, blend.r), BlendScreenf(base.g, blend.g), BlendScreenf(base.b, blend.b))
#define BlendOverlay(base, blend) vec3(BlendOverlayf(base.r, blend.r), BlendOverlayf(base.g, blend.g), BlendOverlayf(base.b, blend.b))
#define BlendSoftLight(base, blend) vec3(BlendSoftLightf(base.r, blend.r), BlendSoftLightf(base.g, blend.g), BlendSoftLightf(base.b, blend.b))
#define BlendHardLight(base, blend) BlendOverlay(blend, base)
#define BlendColorDodge(base, blend) vec3(BlendColorDodgef(base.r, blend.r), BlendColorDodgef(base.g, blend.g), BlendColorDodgef(base.b, blend.b))
#define BlendColorBurn(base, blend) vec3(BlendColorBurnf(base.r, blend.r), BlendColorBurnf(base.g, blend.g), BlendColorBurnf(base.b, blend.b))
#define BlendLinearLight(base, blend) vec3(BlendLinearLightf(base.r, blend.r), BlendLinearLightf(base.g, blend.g), BlendLinearLightf(base.b, blend.b))
#define BlendVividLight(base, blend) vec3(BlendVividLightf(base.r, blend.r), BlendVividLightf(base.g, blend.g), BlendVividLightf(base.b, blend.b))
#define BlendPinLight(base, blend) vec3(BlendPinLightf(base.r, blend.r), BlendPinLightf(base.g, blend.g), BlendPinLightf(base.b, blend.b))
#define BlendHardMix(base, blend) vec3(BlendHardMixf(base.r, blend.r), BlendHardMixf(base.g, blend.g), BlendHardMixf(base.b, blend.b))
#define BlendReflect(base, blend) vec3(BlendReflectf(base.r, blend.r), BlendReflectf(base.g, blend.g), BlendReflectf(base.b, blend.b))
#define BlendGlow(base, blend) BlendReflect(blend, base)
#define BlendPhoenix(base, blend) (min(base, blend) - max(base, blend) + CAST3(1.0))
#define BlendOpacity(base, blend, F, O) mix(base, F(base, blend), O)
#define BlendLinearDodge(base, blend) min(base + blend, CAST3(1.0))
#define BlendLinearBurn(base, blend) max(base + blend - CAST3(1.0), CAST3(0.0))
#define BlendTint(base, blend) (CAST3(max(base.x, max(base.y, base.z))) * blend)

vec3 BlendHue(vec3 base, vec3 blend)
{
	vec3 baseHSL = RGBToHSL(base);
	return HSLToRGB(vec3(RGBToHSL(blend).r, baseHSL.g, baseHSL.b));
}

vec3 BlendSaturation(vec3 base, vec3 blend)
{
	vec3 baseHSL = RGBToHSL(base);
	return HSLToRGB(vec3(baseHSL.r, RGBToHSL(blend).g, baseHSL.b));
}

vec3 BlendColor(vec3 base, vec3 blend)
{
	vec3 blendHSL = RGBToHSL(blend);
	return HSLToRGB(vec3(blendHSL.r, blendHSL.g, RGBToHSL(base).b));
}

vec3 BlendLuminosity(vec3 base, vec3 blend)
{
	vec3 baseHSL = RGBToHSL(base);
	return HSLToRGB(vec3(baseHSL.r, baseHSL.g, RGBToHSL(blend).b));
}

vec3 ApplyBlending(const int blendMode, in vec3 A, in vec3 B, in float opacity)
{
#if BLENDMODE == 1
		return mix(A,BlendDarken(A,B),opacity);
#endif
#if BLENDMODE == 2
		return mix(A,BlendMultiply(A,B),opacity);
#endif
#if BLENDMODE == 3
		return mix(A,BlendColorBurn(A,B),opacity);
#endif
#if BLENDMODE == 4
		return mix(A,BlendSubstract(A,B),opacity);
#endif
#if BLENDMODE == 5
		return min(A, B);
#endif
#if BLENDMODE == 6
		return mix(A,BlendLighten(A,B),opacity);
#endif
#if BLENDMODE == 7
		return mix(A,BlendScreen(A,B),opacity);
#endif
#if BLENDMODE == 8
		return mix(A,BlendColorDodge(A,B),opacity);
#endif
#if BLENDMODE == 9
		return mix(A,BlendAdd(A,B),opacity);
#endif
#if BLENDMODE == 10
		return max(A, B);
#endif
#if BLENDMODE == 11
		return mix(A,BlendOverlay(A,B),opacity);
#endif
#if BLENDMODE == 12
		return mix(A,BlendSoftLight(A,B),opacity);          r.rgb *= g_Brightness;
#endif
#if BLENDMODE == 13
		return mix(A,BlendHardLight(A,B),opacity);          r.a *= g_Alpha;
#endif
#if BLENDMODE == 14
		return mix(A,BlendVividLight(A,B),opacity);         t metallic = g_Metallic;
#endif
#if BLENDMODE == 15
		return mix(A,BlendLinearLight(A,B),opacity);         componentMaps = texSample2D(g_Texture2, v_T
#endif
#if BLENDMODE == 16
		return mix(A,BlendPinLight(A,B),opacity);           _MAP
#endif
#if BLENDMODE == 17
		return mix(A,BlendHardMix(A,B),opacity);
#endif
#if BLENDMODE == 18
		return mix(A,BlendDifference(A,B),opacity);
#endif
#if BLENDMODE == 19
		return mix(A,BlendExclusion(A,B),opacity);           normal = vec3(compressedNormal,
#endif
#if BLENDMODE == 20
		return mix(A,BlendSubstract(A,B),opacity);          al = normalize(normal);
#endif
#if BLENDMODE == 21
		return mix(A,BlendReflect(A,B),opacity);
#endif
#if BLENDMODE == 22
		return mix(A,BlendGlow(A,B),opacity);
#endif
#if BLENDMODE == 23
		return mix(A,BlendPhoenix(A,B),opacity);
#endif
#if BLENDMODE == 24
		return mix(A,BlendAverage(A,B),opacity);
#endif
#if BLENDMODE == 25
		return mix(A,BlendNegation(A,B),opacity);           iplied[0].rgb, f0, roughness, metallic);
#endif
#if BLENDMODE == 26
		return mix(A,BlendHue(A,B),opacity);
#endif
#if BLENDMODE == 27
		return mix(A,BlendSaturation(A,B),opacity);         t += ComputePBRLight(normal, vec3(v_Light0Di
#endif
#if BLENDMODE == 28
		return mix(A,BlendColor(A,B),opacity);
#endif
#if BLENDMODE == 29
		return mix(A,BlendLuminosity(A,B),opacity);             f0, roughness, metallic);
#endif
#if BLENDMODE == 30
		return mix(A,BlendTint(A,B),opacity);
#endif
#if BLENDMODE == 31
		return A + B * opacity;
#endif
		return mix(A,BlendNormal(A,B),opacity);
}
	color.rgb = saturate(ambient + light) + (light *
// end of included from file common_blending.h
	
	//float overbright = saturate((length(light) - 2
void main() {
	vec4 color = texSample2D(g_Texture0, v_TexCoord.xy);        overbright);
#else
#ifndef VERSION
	color.rgb *= g_Brightness;
	color.a *= g_UserAlpha;
#else
	color.rgb *= g_Color;
	color.a *= g_Alpha;
#endif

#if LIGHTING || REFLECTION
	float metallic = g_Metallic;
	float roughness = g_Roughness;
#if REFLECTION_MAP
#if PBRMASKS
	vec3 componentMaps = texSample2D(g_Texture2, v_TexCoord.zw).rgb;
#endif
	vec2 bitangent = normalize(v_Bitangent.xy);
#if METALLIC_MAP
	metallic = componentMaps.x;
#endif
	float fresnelTerm = max(0.001, dot(normal, vec3(
#if ROUGHNESS_MAP
	roughness = componentMaps.y;
#endif
_Screen.z, 0.25);
#if NORMALMAP
	vec2 compressedNormal = texSample2D(g_Texture1, v_TexCoord.xy).xy * 2.0 - 1.0;
	vec3 normal = vec3(compressedNormal,
		sqrt(saturate(1.0 - compressedNormal.x * compressedNormal.x - compressedNormal.y * compressedNormal.y)));               gent * normal.x + bitangent * no
	normal = normalize(normal);
#else
	vec3 normal = normalize(v_Normal);
#endif
	vec3 f0 = CAST3(0.04);
	f0 = mix(f0, color.rgb, metallic);
	reflectionColor = pow(max(CAST3(0.001), reflecti
	// Using the actual view vector is ugly for ortho rendering
	vec3 normalizedViewVector = vec3(0, 0, 1); //normalize(v_ViewDir);
#endif
	gl_FragColor = color;
#if LIGHTING
	vec3 light = ComputePBRLight(normal, v_Light0DirectionL3X.xyz, normalizedViewVector, color.rgb, g_LightsColorPremultiplied[0].rgb, f0, roughness, metallic);
	light += ComputePBRLight(normal, v_Light1DirectionL3Y.xyz, normalizedViewVector, color.rgb, g_LightsColorPremultiplied[1].rgb, f0, roughness, metallic);
	light += ComputePBRLight(normal, v_Light2DirectionL3Z.xyz, normalizedViewVector, color.rgb, g_LightsColorPremultiplied[2].rgb, f0, roughness, metallic);
	light += ComputePBRLight(normal, vec3(v_Light0DirectionL3X.w, v_Light1DirectionL3Y.w, v_Light2DirectionL3Z.w),
		normalizedViewVector, color.rgb,
		vec3(g_LightsColorPremultiplied[0].w, g_LightsColorPremultiplied[1].w, g_LightsColorPremultiplied[2].w),                7655192                         
		f0, roughness, metallic);
	vec3 ambient = max(CAST3(0.001), g_LightAmbientColor) * color.rgb;
ing to normal folder storage
	// Why disable HDR with this??
	//vec3 tmp = ambient + light;
#if HDR
	float lightLen = length(light);
	float overbright = (saturate(lightLen - 2.0) * 0.1) / max(0.01, lightLen);
	color.rgb = saturate(ambient + light) + (light * overbright);
	
	//float overbright = saturate((length(light) - 2.0) * 0.1);
	//color.rgb = saturate(ambient + light) + (normalize(tmp) * overbright);
#else
	color.rgb = ambient + light;
#endif
	// Disables HDR correction
	//color.rgb = ambient + light;
	//color.rgb = vec3(length(g_LightsColorPremultiplied[0].xyz) / 2000.0, 0, 0);
#endif
I"
#if REFLECTION && NORMALMAP
	float reflectivity = g_Reflectivity;
Compiled source code:
#if REFLECTION_MAP
	reflectivity *= componentMaps.z;
#endif
#define lowp
	vec2 tangent = normalize(v_Tangent.xy);
	vec2 bitangent = normalize(v_Bitangent.xy);
	vec2 screenUV = (v_ScreenPos.xy / v_ScreenPos.z) * 0.5 + 0.5;
#define CAST3(x) (vec3(x))
	float fresnelTerm = max(0.001, dot(normal, vec3(0, 0, 1))); 3X3(x) (mat3(x))
#if PLATFORM_ANDROID
	normal.xy = normalize(normal.xy) * vec2(0.25 / g_Screen.z, 0.25);
#else
	// Make consistent on X since the width is usually more variable (multi monitors) - bad for phones tho
	normal.xy = normalize(normal.xy) * vec2(0.15, 0.15 * g_Screen.z);
#endif
	screenUV += (tangent * normal.x + bitangent * normal.y) * fresnelTerm;

	vec3 reflectionColor = texSample2DLod(g_Texture3, screenUV, roughness * g_Texture3MipMapInfo).rgb;
	reflectionColor = reflectionColor * (1.0 - fresnelTerm) * reflectivity;
	reflectionColor = pow(max(CAST3(0.001), reflectionColor), CAST3(2.0 - metallic));

	color.rgb += saturate(reflectionColor);
#endif

	gl_FragColor = color;

#if BLENDMODE
	vec2 screenCoord = v_ScreenCoord.xy / v_ScreenCoord.z * vec2(0.5, 0.5) + 0.5;
	vec4 screen = texSample2D(g_Texture4, screenCoord);

	gl_FragColor.rgb = ApplyBlending(BLENDMODE, screen.rgb, gl_FragColor.rgb, gl_FragColor.a);
	gl_FragColor.a = screen.a;
#endif
}

[1]    8701 IOT instruction (core dumped)  wallengine 2463435445
```

**Desktop:**
 
- OS: Gentoo Linux (Linux 5.17.1-zen1-x86_64)
 - Window Manager: BSPWM (via lightdm-gtk-greeter)

**Additional context**

Some wallpapers work fine, but most of them return that "M_PI" undeclared issue.
Compiled yesterday with latest commit 46ea8e8 (main)

**Comment by Arna13** [comment]
I can confirm I have the same problem with 2685564485 as in #100 too. I wonder what Almamu's setup is as they claim it works on their end.

Regarding the GPU, I have a MUXed laptop where I can connect the internal display directly to the intel iGPU and form an Optimus setup or connect the Nvidia card directly to the display and completely disable the iGPU.

At the moment of testing when opening this issue I was using my Nvidia card (RTX3060 Max-Q) with the proprietary drivers (515.48.07). I am now using the Optimus setup with the Nvidia card turned off (with vfio_pci loaded on it) and Xorg running on intel graphics drivers and I get the same error, so it is probably not a GPU specific problem.

@mihawk90 Would you mind sharing your kernel version? Maybe it has to do with that, since I am using zen-kernel 5.17.1 and I believe Fedora uses a pretty bleeding edge kernel too.

**Comment by mihawk90** [comment]
Fedora 36 is on 5.17.12

**Comment by Almamu** [comment]
> Maybe this is useful: [GNU Mathematical Constants](https://www.gnu.org/software/libc/manual/html_node/Mathematical-Constants.html#:~:text=M_PI,Pi%20divided%20by%20two.)
> 
> Edit: Seems to be part of libc (math.h), I will look at it when I have some free time.

The error comes from the shader compiler, so libc shouldn't have anything to do with it. I can launch all the backgrounds in here, but it looks like this has to do with changes on the wallpaper engine assets shipped with the backgrounds, as using an old version (which is what I was using) works just fine with all the backgrounds, but once I switch to the live, steam version, I cannot launch any background. I'm investigating this a bit further, but looks like there's some new definitions that have to be added to the shaders :/ I'll keep y'all updated

**Comment by Almamu** [comment]
Should be fixed with the last commit (cf2e31a6062961c706f3db56c7d0264c656d786d) please let me know if it's still crashing like this.
